targets/TARGET_NXP/USBHAL_LPC17.cpp

```diff
@@ -411,7 +411,7 @@ void USBPhyHw::init(USBPhyEvents *events)
     LPC_PINCON->PINSEL4 |= 0x00040000;
 
     // Connect must be low for at least 2.5uS
-    ThisThread::sleep_for(300);
+    wait_us(300000);
```

Comment:
I previously only read a comment about 2.5uS and we had there 300 ms delay? This is quite a long delay. Wouldn't be sufficient to just a spin for 1ms?

Reply:
900 should be sufficient based on the comment

Reply:
> 900 should be sufficient based on the comment

Yes 900us should be sufficient. Previously it was 300ms, so given that much.
Updated the pull request with 900us.

Reply:
If interrupts are disabled, shouldn't really be waiting for 900us either. That's enough to cause many characters of serial loss.

But why do you think interrupts are disabled? Where is this being called from?

And where did "900us" come from? Only comment I can see says "2.5us".

Reply:
Have a look at issue number:
https://github.com/ARMmbed/mbed-os/issues/13203
Earlier delay given was 300ms. As mentioned in the comments, i given 900us.

Reply:
Issue should be linked in the description - presumably as "fixes #13203".

That `USBDevice::lock()` is pretty heavy. Most standard drivers use a mutex, not a critical section. Doubt it's easy to change though.

> As mentioned in the comments, i given 900us.

Which comment, where? I don't see anything in 13203. Earliest reference I can find is from @0xc0170 [here](https://github.com/ARMmbed/mbed-os/pull/13206#discussion_r447642838)

Reply:
> Issue should be linked in the description - presumably as "fixes #13203".
> 
> That `USBDevice::lock()` is pretty heavy. Most standard drivers use a mutex, not a critical section. Doubt it's easy to change though.
> 
> > As mentioned in the comments, i given 900us.
> 
> Which comment, where? I don't see anything in 13203. Earliest reference I can find is from @0xc0170 [here](https://github.com/ARMmbed/mbed-os/pull/13206#discussion_r447642838)

Yes, mutex should be used in place lock() but this had been done long time back.

Reply:
Should I implement mutex solution for this and give a pull request?

Reply:
I'll ask Russ why USBDevice lock implements critical section, will get back

Reply:
Thinking about this. as the comment states 2.5us minimal, therefore waiting here 3us should be sufficient (there could be just typo and 300ms applied instead earlier).

The USB Device should be interrupt safe, thus locking the way it is. I propose to wait 5us to be on the safe side (stating that in the comment).

Reply:
> Thinking about this. as the comment states 2.5us minimal, therefore waiting here 3us should be sufficient (there could be just typo and 300ms applied instead earlier).
> 
> The USB Device should be interrupt safe, thus locking the way it is. I propose to wait 5us to be on the safe side (stating that in the commit message to be clear).

I will change the delay to 5us then.

Reply:
> I'll ask Russ why USBDevice lock implements critical section, will get back

Ok martin, Could you please update here, if mutex needs to be implemented for this.